src/main/kotlin/sh/hsp/techtree/TreeModelValidator.kt

```diff
@@ -0,0 +1,36 @@
+package sh.hsp.techtree
+import java.io.IOException
+import java.util.stream.Collectors
+
+class TreeModelValidator {
+    companion object {
```

Comment:
Ogólnie używanie statycznych metod tutaj, które nie są konstruktorami to jest no no.
Działa tutaj, więc może zostać.

Normalnie to powinien być obiekt, który też jest wstrzykiwany przez konstruktor.